processing/src/main/java/io/druid/query/extraction/RegexDimExtractionFn.java

```diff
@@ -34,28 +35,59 @@
 public class RegexDimExtractionFn extends DimExtractionFn
 {
   private static final byte CACHE_TYPE_ID = 0x1;
+  private static final byte CACHE_KEY_SEPARATOR = (byte) 0xFF;
 
   private final String expr;
   private final Pattern pattern;
+  private final boolean injective;
+  private final boolean replaceMissingValues;
+  private final String replaceMissingValuesWith;
 
   @JsonCreator
   public RegexDimExtractionFn(
-      @JsonProperty("expr") String expr
-  )
+      @JsonProperty("expr") String expr,
+      @JsonProperty("replaceMissingValues") Boolean replaceMissingValues,
+      @JsonProperty("replaceMissingValuesWith") String replaceMissingValuesWith,
+      @JsonProperty("injective") Boolean injective
```

Comment:
i don't see why we need this property ? it  relevant for function lookup but i don't think it is useful for regex. If you have any use case please let me know.